### ISSUE TITLE
Implement Sparsemax Activation and Loss

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/addons#21]
Per the [RFC](https://github.com/tensorflow/community/blob/393dfc9e03dd4150a839de6357a26e0259aa97d6/rfcs/20181214-move-to-addons.md), we need to move [sparsemax](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/sparsemax) from contrib to addons:

- [X] Moving sparsemax to addons/activations and modifying test cases to run in TF2.
- [X] Restructuring sparsemax_loss in an OO format by inheriting from base Keras Loss, modifying the code to match those APIs, and modifying test cases to run in TF2.

https://arxiv.org/abs/1602.02068

**Comment by facaiy** [comment]
@AndreasMadsen Hi, Andreas. As the author of [sparsemax](https://github.com/tensorflow/tensorflow/commits?author=AndreasMadsen), would you like to migrate it from contrib to addons? That would be greatly helpful :-)

**Comment by AndreasMadsen** [comment]
Yep. I will give it a go. Perhaps next week.

**Comment by facaiy** [comment]
Great!
